### ISSUE TITLE
Inconsistent error messages when applying function to incompatible types

### ISSUE THREAD

**Issue by cleishm** [neo4j/neo4j#11980]
### Actual behavior

```
neo4j> CREATE (n:Test {loc: point({x:22, y:44})}) RETURN 2 + n.loc;
<interactive>:0:0: error: Don't know how to add `Long(2)` and `point({x: 22.0, y: 44.0, crs: 'cartesian'})`
neo4j> CREATE (n:Test {loc: point({x:22, y:44})}) RETURN 2 * n.loc;
<interactive>:0:0: error: Don't know how to Multiply(NodeProperty(0,2),{  AUTOINT2}) `Long(2)` with `point({x: 22.0, y: 44.0, crs: 'cartesian'})`
```
```
neo4j> CREATE (n:Test {loc: point({x:22, y:44})});
Created 1 node, set 1 property, added 1 label
0 rows returned in 2ms (rendered after 5ms)
neo4j> MATCH (n:Test) RETURN 2 + n.loc;
+------------------------------------------------------------------------------+
|                                                                              |
+------------------------------------------------------------------------------+
<interactive>:0:0: error: Don't know how to add `2` and `point({x: 22.0, y: 44.0, crs: 'cartesian'})`
neo4j> MATCH (n:Test) RETURN 2 * n.loc;,
+------------------------------------------------------------------------------+
|                                                                              |
+------------------------------------------------------------------------------+
<interactive>:0:0: error: Cannot multiply PointValue and Long
```

```
neo4j> CREATE (n:Test2 {s: ""}) RETURN 2 * n.s;
<interactive>:0:0: error: Don't know how to Multiply(NodePropertyLate(0,s),{  AUTOINT1}) `Long(2)` with `String("")`
neo4j> CREATE (n:Test2 {s: ""});
Created 1 node, set 1 property, added 1 label
0 rows returned in 8ms (rendered after 11ms)
neo4j> MATCH (n:Test2) RETURN 2 * n.s;
+------------------------------------------------------------------------------+
|                                                                              |
+------------------------------------------------------------------------------+
<interactive>:0:0: error: Cannot multiply  and Long
```
(note the extra space and missing type name for the string)

### Expected behavior

Error messages should be consistent. E.g.:

* "Cannot add Long and *Point*"
* "Cannot multiply Long and *Point*"
* "Cannot add Long and *Point*"
* "Cannot multiply Long and *Point*"
* "Cannot multiply Long and String"
* "Cannot multiply Long and String"

(Note: for consistency with other types, it should be Point and not PointValue)

**Comment by SaschaPeukert** [comment]
Thanks for your feedback, I will make sure it reaches the right people

**Comment by sherfert** [comment]
This is fixed in 3.4.7
